### DISCUSSION TITLE
MacOS Monteray 12.1 File Browse Failure

### DISCUSSION THREAD

**Discussion by wak932** [airsdk/Adobe-Runtime-Support#1518] [General]
Just editing the original discussion ticket - whilst the new build fixed the main error it has caused another.. you cannot select the default folder of a browseForDirectory
https://github.com/airsdk/Adobe-Runtime-Support/issues/1522

--------
Original discussion

Since upgrading to MacOS 12.1 we can only call one .browseForOpen() file dialog. That first one works fine.
Any subsequent attempts to browse the file system opens the dialog but the app (and the dialog) hang with no error info.
This is obviously a pretty critical problem.
Can you confirm this and let us know what might be happening?
Thanks
.jez

**Comment by wak932** [comment]
To show you how easy it is to recreate, try this class.
Click button once - select/cancel/whatever out of the dialog
Click button again... hangs.
remember this is 12.1

package {
import flash.display.MovieClip;
import flash.display.Sprite;
import flash.events.MouseEvent;
import flash.filesystem.File;

public class TestClass extends MovieClip{

    private var browseFile:File;
    public function TestClass() {
        var but:Sprite = new Sprite();
        but.graphics.beginFill(0x888888);
        but.graphics.drawRect(100,100,100,50);
        but.addEventListener(MouseEvent.CLICK, browseForOpen);
        addChild(but);
    }

    private function browseForOpen(event:MouseEvent):void {
        browseFile = new File();
        browseFile.browseForOpen('browse test');
    }
}
}

**Comment by wak932** [comment]
Just seen the 731 release from Yesterday is supposed to fix - so ignore me - will check the lists before posting!

**Comment by twhamilton** [comment]
FYI this is related to: 

https://github.com/airsdk/Adobe-Runtime-Support/issues/1509
731 fixed the open twice issue but there are other issues still with file browse dialogs.
